modules/renter/files.go

```diff
@@ -176,30 +178,41 @@ func (r *Renter) FileList() []modules.FileInfo {
 // file must exist, and there must not be any file that already has the
 // replacement nickname.
 func (r *Renter) RenameFile(currentName, newName string) error {
-	return errors.New("renaming is disabled")
-
-	/*
-		lockID := r.mu.Lock()
-		defer r.mu.Unlock(lockID)
+	lockID := r.mu.Lock()
+	defer r.mu.Unlock(lockID)
 
-		// Check that the currentName exists and the newName doesn't.
-		file, exists := r.files[currentName]
-		if !exists {
-			return ErrUnknownNickname
-		}
-		_, exists = r.files[newName]
-		if exists {
-			return ErrNicknameOverload
-		}
+	// Check that currentName exists and newName doesn't.
+	file, exists := r.files[currentName]
+	if !exists {
+		return ErrUnknownNickname
+	}
+	_, exists = r.files[newName]
+	if exists {
+		return ErrNicknameOverload
+	}
 
-		// Do the renaming.
-		file.name = newName
-		r.saveFile(file)
-		delete(r.files, currentName)
-		r.files[newName] = file
+	// Modify the file and save it to disk.
+	file.mu.Lock()
+	file.name = newName
+	err := r.saveFile(file)
+	file.mu.Unlock()
+	if err != nil {
+		return err
+	}
 
-		r.save()
-		return nil
+	// Update the entries in the renter.
+	delete(r.files, currentName)
+	r.files[newName] = file
+	err = r.save()
+	if err != nil {
+		return err
+	}
 
-	*/
+	// Delete the old .sia file.
+	// NOTE: proper error handling is difficult here. For example, if the
+	// removal fails, should the entry in r.files be preserved? For now we will
+	// keep things simple, but it is important that our approach feels
+	// intuitive/unsurprising and doesn't put the user's data at risk.
```

Comment:
restarting the client will cause it to read in any sia files in the directory structure